src/Core/PostgreSQL/Connection.cpp

```diff
@@ -59,8 +59,16 @@ void Connection::updateConnection()
     if (connection)
         connection->close();
 
-    /// Always throws if there is no connection.
-    connection = std::make_unique<pqxx::connection>(connection_info.connection_string);
+    try
+    {
+        /// Always throws if there is no connection.
+        connection = std::make_unique<pqxx::connection>(connection_info.connection_string);
+    }
+    catch (const pqxx::broken_connection & e)
+    {
+        LOG_ERROR(log, "Unable to update connection: {}", e.what());
+        throw;
```

Comment:
But after this line, the connection is also unset.

Reply:
connection = null is catched and rethrowed, so that in PoolWithFailover::get() connection is recycled and scheduled to reconnect after several trials

Reply:
My bad, the exception is caught at upper level already. Close this PR.